### PR TITLE
feat: punchout identity provider

### PR DESCRIPTION
**Pull request by Eisie96**
## PR Type

[x] Feature

## What Is the Current Behavior?

OCI Punchout Sessions gets lost after Reload.

Issue Number: Closes #

## What Is the New Behavior?

OCI Punchout Sessions is available after Reload. Therefore is the punchout guard refactored to a new identity provider, which saves the actual basketId to the session storage. Furthermore it is possible to override the actual identity provider on specific routes. 

## Does this PR Introduce a Breaking Change?

[ ] Yes
[ ] No

## Other Information

[AB#70891](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/70891)

### CODE REVIEW

File: src/app/extensions/punchout/identity-provider/punchout-identity-provider.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                  sameSite: 'None',
+                  secure: true,
+                });
+                this.cookiesService.put('punchout_BasketID', data.basketId, {
+                  sameSite: 'None',
+                  secure: true,
+                });
+              }),
+              // use the basketId basket for the current PWA session (instead of default current basket)
+              // TODO: if load basket error (currently no error page) -> logout and do not use default 'current' basket
```
Comment (MaxKless) [inline src/app/extensions/punchout/identity-provider/punchout-identity-provider.ts:112]: Are you planning on handling these TODOs? Or is that for a later time?

**Review by MaxKless — CHANGES_REQUESTED** [review]

File: nginx/entrypoint.sh [hunk 2]
```diff
@@ -26,6 +28,14 @@ then
   fi
 fi
 
+if [ -z "$OVERRIDE_IDENTITY_PROVIDERS_SOURCE" ]
+then
+  if [ -n "$OVERRIDE_IDENTITY_PROVIDERS" ]
```
[the remark this answers is not in the document]
Comment (Eisie96) [reply nginx/entrypoint.sh:33]: -n stands for not null. The entryproypoint checks only the case, if the environment variable OVERRIDE_IDENTITY_PROVIDERS was set. If so, OVERRIDE_IDENTITY_PROVIDERS_SOURCE gets the yaml parsed value from that variable.

**Review by MaxKless — DISMISSED** [review]
Great job!

**Comment by MaxKless** [comment]
Sadly, I found a bug when opening two windows with OCI punchout when using a static punchout identity provider.
Steps to repeat:
- start an OCI punchout session 
- open a second window of the PWA and open another OCI punchout session
- add different items to the cart so you can easily distinguish between the two sessions
- reload one window (sometimes twice or more)
- open the other window and see the confusing behavior


When building the PWA with `ng serve`, the other window will be logged out.
When building the PWA with `docker-compose`, the other window will be logged out and a tiny sign-in window appears:
image

**Pull request merged**